client/src/components/SentimentAnalysis/SentimentAnalysis.js

```diff
@@ -80,8 +82,24 @@ const SentimentAnalysis = props => {
       : locationState.ad.content;
   };
 
+  /**
+   * Generates a PNG image of the current result page and
+   * automatically saves it to user's local machine
+   * @returns {Void} doesn't return anything
+   */
+  const handleDownload = () => {
```

Comment:
Maybe a better/more descriptive name would be `downloadAsImage`. Will leave up to you.